framework/Models/PostProcessors/ParetoFrontierPostProcessor.py

```diff
@@ -80,24 +71,20 @@ def _handleInput(self, paramInput):
       @ In, paramInput, ParameterInput, the already-parsed input.
       @ Out, None
     """
-    PostProcessor._handleInput(self, paramInput)
-    costID  = paramInput.findFirst('costID')
-    self.costID  = costID.value
-    self.invCost = costID.parameterValues['inv']
-
-    valueID = paramInput.findFirst('valueID')
-    self.valueID = valueID.value
-    self.invValue = valueID.parameterValues['inv']
-
-    costLimit  = paramInput.findFirst('costLimit')
-    if costLimit is not None:
-      self.costLimit  = costLimit.value
-      self.costLimitType = costLimit.parameterValues['type']
-
-    valueLimit = paramInput.findFirst('valueLimit')
-    if valueLimit is not None:
-      self.valueLimit = valueLimit.value
-      self.valueLimitType = valueLimit.parameterValues['type']
+    self.objectives = {}
+
+    for child in paramInput.subparts:
+      if child.getName() == 'objective':
+        self.objectives[child.value]={}
+        if child.parameterValues['goal'] in ['min','max']:
```

Comment:
You can use InputTypes.makeEnumType in getInputSpecification to handle this, in that case, you do not need to check the goal input here.

Reply:
good point: added enumType and removed checks